src/platform/nrf_pca10001/lib.rs

```diff
@@ -0,0 +1,59 @@
+#![crate_name = "platform"]
+#![crate_type = "rlib"]
+#![no_std]
+
+extern crate drivers;
+extern crate hil;
+extern crate nrf51822;
+
+pub struct Firestorm {
+    gpio: &'static drivers::gpio::GPIO<[&'static hil::gpio::GPIOPin; 32]>,
+}
+
+impl Firestorm {
+    pub unsafe fn service_pending_interrupts(&mut self) {
+    }
+
+    pub unsafe fn has_pending_interrupts(&mut self) -> bool {
+        // FIXME: The wfi call from main() blocks forever if no interrupts are generated. For now,
+        // pretend we have interrupts to avoid blocking.
+        true
+    }
+
+    #[inline(never)]
+    pub fn with_driver<F, R>(&mut self, driver_num: usize, f: F) -> R where
+            F: FnOnce(Option<&hil::Driver>) -> R {
+        match driver_num {
+            1 => f(Some(self.gpio)),
+            _ => f(None)
+        }
+    }
+}
+
+pub unsafe fn init<'a>() -> &'a mut Firestorm {
+    use core::mem;
+    use nrf51822::gpio::PA;
+
+    static mut FIRESTORM_BUF : [u8; 1024] = [0; 1024];
+    static mut GPIO_BUF : [u8; 1024] = [0; 1024];
+
+    let gpio : &mut drivers::gpio::GPIO<[&'static hil::gpio::GPIOPin; 32]> = mem::transmute(&mut GPIO_BUF);
+
+    *gpio = drivers::gpio::GPIO::new([
+        &mut PA[0], &mut PA[1], &mut PA[2], &mut PA[3],
```

Comment:
The `mut`s here are actually discarded, since the GPIO driver just takes a list immutable static pointers.